api/v1/cluster_types.go

```diff
@@ -45,6 +47,10 @@ type ClusterStatus struct {
 	NodeProvisionStatus string `json:"node_provision_status,omitempty"`
 }
 
+func (c Cluster) Key() string {
+	return "clsuter" + "-" + strconv.Itoa(c.ID) + "-" + c.Metadata.Name
```

Comment:
Maybe we should use workspace + name as the key?

Reply:
Fix